### ISSUE TITLE
False negative for Markdown anchors containing parentheses

### ISSUE THREAD

**Issue by sohang3112** [tcort/markdown-link-check#593]
### Have you checked that the bug is not already fixed?

Yes

### Summary

Internal anchor link is falsely reported as broken when the linked markdown heading contains a bracket. The same link works correctly in Github.

Current workaround is to remove brackets from the heading, then the link is correctly reported live by `markdown-link-check`.

### Affected version

3.15

### Steps to reproduce the bug

1. Create a test markdown file with contents:

```markdown
# Maximum Likelihood Estimator (MLE)

Link: [link](#maximum-likelihood-estimator-mle)
```

2. Run `markdown-link-check` on this markdown file:

```bash
$ markdown-link-check -v README.md  

FILE: README.md 
...
 [✖] #maximum-likelihood-estimator-mle → Status: 404

8 links checked.

ERROR: 1 dead link found!
```

### Expected behavior

The link should be reported as valid.

### Actual behavior

The link is reported as broken.

### Can you fix it?

Yes
